### PR TITLE
Add-publishers #59

### LINKED ISSUE #59: Add publishers
Many books, especially recently recorded, should specify publisher. There can be multiple publishers for the same book, for example https://audiobooks.by/books/aposhniaia-kniga-pana-a has 2 publishers: Янушкевіч і Вясна. 

Publisher should probably be set on Narration level, not Book leve. The reason is that same book can be recorded by different publishers. It becomes potentially tricky in cases like: paper book published by A, then it was recorded by two different publishers B and C. Then book has two narrations. The first narration will have publishers A and B. The second - A and C.

### PR DESCRIPTION
**Pull request by frombrest**
Resolves #59

### CODE REVIEW

File: books/views/publisher.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+        context = {
+            'publisher': publisher,
+            'narrations': narrated_books,
+        }
+
+        return render(request, 'books/publisher.html', context)
+
+    else:
+        pass  # TODO: implement 404 page
```
Comment (nbeloglazov) [inline books/views/publisher.py:31]: Let's use `get_object_or_404` here as we do for books: https://github.com/belaudiobooks/website/blob/main/books/views/book.py#L12 I don't know why we use this different pattern for Person view.

File: templates/partials/_publisher.html [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{% load static %}
+{% load books_extras %}
```
Comment (nbeloglazov) [inline templates/partials/_publisher.html:2]: I don't think we need to load static or books_extras for this page as they are not used?

File: books/views/publisher.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+from books.models import BookStatus, Publisher
+
+
+def publisher_detail(request: HttpRequest, slug: str) -> HttpResponse:
+    """Detailed publisher page"""
+
+    publisher = Publisher.objects.filter(slug=slug).prefetch_related('narrations').first()
+
+    if publisher:
+        narrations = publisher.narrations.all().filter()
```
Comment (nbeloglazov) [inline books/views/publisher.py:17]: .filter() is reduntant

File: books/views/publisher.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+        narrated_books = [
+            item.book for item in narrations
+            if item.book.status == BookStatus.ACTIVE
+        ]
+
+        context = {
+            'publisher': publisher,
+            'narrations': narrated_books,
```
Comment (nbeloglazov) [inline books/views/publisher.py:25]: Let's call it `books` rather that `narrations`. Given that it's a list of Book models `books` better reflects what it is.

File: templates/books/publisher.html [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+{% block og_image %}{% spaceless %}
+    {% if publisher.logo %}
+        {{ publisher.logo.url }}
+    {% else %}
+        {{ block.super }}
+    {% endif %}
+{% endspaceless %}{% endblock %}
+
+{% block description %}
+  На гэтай старонке вы знойдзеце ўсе кнігі выданыя альбо агучаныя выдавецтвам {{publisher.name}}.
```
Comment (nbeloglazov) [inline templates/books/publisher.html:16]: grammar: старонЦЫ

**Review by nbeloglazov — CHANGES_REQUESTED** [review]
Looks good! Left few comments. Few should also add webdriver tests. But they are currently rely on database containing actual publishers. So I first need to merge this PR, create few publishers and then we can write tests. Also filed #90 for migrating tests.

File: templates/books/publisher.html (hunk 5 above)
Comment (frombrest) [reply templates/books/publisher.html:16]: That is something I wanted to clarify:
старонКЕ ? старонЦЫ
аўдыЯкнігі ? аўдыЁкнігі

Comment (nbeloglazov) [reply templates/books/publisher.html:16]: старонЦЫ: https://slounik.org/search?dict=&search=%D1%81%D1%82%D0%B0%D1%80%D0%BE%D0%BD%D0%BA%D0%B0

І мы выкарыстоўваем аўдыЯкнігі. Шмат хто выкарыстоўвае Ё, але мы вырашылі Я, т.к. гэта больш адпавядае сучаснаму правапісу.

**Comment by nbeloglazov** [comment]
Ah, and we also should add links to all publishers to the sitemap page: https://github.com/belaudiobooks/website/blob/main/books/views/support.py#L110

**Comment by frombrest** [comment]
> Ah, and we also should add links to all publishers to the sitemap page: https://github.com/belaudiobooks/website/blob/main/books/views/support.py#L110

Added

**Review by nbeloglazov — APPROVED** [review]

**Pull request merged**